riocli/device/model.py

```diff
@@ -76,3 +104,54 @@ def validate(data):
         """
         schema = load_schema('device')
         schema.validate(data)
+
+    def create_hwil_device(self) -> str:
+        """Create a new hardware-in-the-loop device."""
+        os = self.spec['virtual']['os']
+        codename = self.spec['virtual']['codename']
+        arch = self.spec['virtual']['arch']
+        labels = self.prepare_device_labels()
+        device_name = self.metadata['name'] + self.spec['virtual']['product'] + self.spec.labels.user
+        client = new_hwil_client()
+        device_id = find_device_id(client, device_name)
+        if device_id != "HWIL device not found":
+            return device_name
+        print(click.style(f'Device configuration = {os}:{codename}:{arch}', fg='cyan', bold=True))
+
+        try:
+            client.create_device(device_name, arch, os, codename, labels)
+            click.secho(f'Device {device_name} created successfully.', fg='green')
+            return device_name
+        except Exception as e:
+            click.secho(f'Failed to create device: {str(e)}', fg='red')
+            raise SystemExit(1)
+
+    def execute_onboard_command(self, device_name: str, onboard_command: str) -> None:
+        """Execute the onboard command on a hardware-in-the-loop device."""
+        client = new_hwil_client()
+        device_id = find_device_id(client, device_name)
+        client.poll_till_device_ready(device_id, sleep_interval=5, retry_limit=3)
+        try:
+            code, stdout, stderr = execute_command(client, device_id, onboard_command)
+            sys.stdout.write(stdout)
+            sys.stderr.write(stderr)
+            sys.exit(code)
```

Comment:
We don't have to do this here. Instead, if exit code is not 0, raise an exception.